plugins/woocommerce/src/Admin/API/Patterns.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\API;
+
+use Automattic\WooCommerce\Blocks\BlockPatterns;
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\Patterns\PTKClient;
+use Automattic\WooCommerce\Blocks\Patterns\PTKPatternsStore;
+use Automattic\WooCommerce\StoreApi\Exceptions\RouteException;
+use Exception;
+use WP_Error;
+use WP_HTTP_Response;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product controller
+ *
+ * @internal
+ */
+class Patterns {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/patterns',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_pattern' ),
+					'permission_callback' => function () {
```

Comment:
Previously we have used `array( Middleware::class, 'is_authorized' )` in the other APIs. Can you explain the difference please?

Reply:
Gotcha, thank you!

Reply:
That said, its a very generic class and method but its purpose is very specific (for checking for authorized ai connections). Unrelated to this PR though